org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/dtd/participants/diagnostics/LSPXML11DTDProcessor.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.dtd.participants.diagnostics;
+
+import org.apache.xerces.impl.Constants;
+import org.apache.xerces.impl.XMLEntityManager;
+import org.apache.xerces.impl.XMLErrorReporter;
+import org.apache.xerces.impl.dtd.XML11DTDProcessor;
+import org.apache.xerces.impl.msg.XMLMessageFormatter;
+import org.apache.xerces.xni.parser.XMLEntityResolver;
+import org.eclipse.lemminx.extensions.xerces.LSPXMLEntityManager;
+
+/**
+ * The Xerces {@link XML11DTDProcessor} doesn't give the capability to
+ * customize the XML entity manager.
+ * 
+ * <p>
+ * This class extends {@link XML11DTDProcessor} and customize the XML entity
+ * manager with the {@link LSPXMLEntityManager} which takes care of download

```

Comment:
which handles the "download external entities" setting.

Reply:
My comment was wrong because it manages any error when download is processed. So I think 

`any remote resource download errors.`

should be better, what do you think about that?